Port to Yocto scarthgap [Part2]

quaresmajose opened the pull request:
The full PR was submitted https://github.com/foundriesio/meta-lmp/pull/1532 but it was very big and we decided to break it down into smaller parts to facilitate the review process.

These small parts have not been tested at all and only the complete PR has been, so it is quite possible that they will not work separately as they are submitted now.

The full list of changes that have been tested are at https://github.com/foundriesio/meta-lmp/tree/scarthgap-next


**Rebased on top of https://github.com/foundriesio/meta-lmp/pull/1538**

ricardosalveti commented on meta-lmp-base/recipes-containers/docker-compose/docker-compose_git.bb:
```diff
@@ -0,0 +1,80 @@
+"
+
+# Specify the first two important SRCREVs as the format
+SRCREV_FORMAT="compose_survey"
```
Missing extra space before/after =.

ricardosalveti commented on meta-lmp-base/recipes-containers/docker-compose/docker-compose_git.bb:
```diff
@@ -0,0 +1,80 @@
+
+do_install() {
+        #install -d "${D}${BIN_PREFIX}/bin"
+        #install -m 755 "${S}/src/import/bin/docker-compose" "${D}${BIN_PREFIX}/bin"
```
These lines can be removed.

quaresmajose replied:
fixed

ricardosalveti commented on meta-lmp-base/recipes-containers/docker-compose/docker-compose_git.bb:
```diff
@@ -0,0 +1,80 @@
+
+}
+
+RDEPENDS:${PN} += " docker"
```
No need for the extra space when using +=.

quaresmajose replied:
fixed

ricardosalveti commented on meta-lmp-base/recipes-containers/docker-compose/docker-compose_git.bb:
```diff
@@ -0,0 +1,80 @@
+        #install -d "${D}${BIN_PREFIX}/bin"
+        #install -m 755 "${S}/src/import/bin/docker-compose" "${D}${BIN_PREFIX}/bin"
+
+	# commonly installed to: /usr/lib/docker/cli-plugins/
```
You can add that it is installed there because compose is now a docker plugin and not a separated exec by default.

quaresmajose replied:
fixed

ricardosalveti commented on meta-lmp-base/recipes-containers/containerd/containerd-opencontainers/0001-Makefile-allow-GO_BUILD_FLAGS-to-be-externally-speci.patch:
```diff
@@ -1,35 +0,0 @@
-From 474b93f8f643651144304a6971ef476add7f06f8 Mon Sep 17 00:00:00 2001
-From: Bruce Ashfield <bruce.ashfield@gmail.com>
```
Same.

quaresmajose replied:
have that on https://github.com/foundriesio/meta-lmp/pull/1444. fixed now

ricardosalveti commented on meta-lmp-base/recipes-containers/docker-compose/docker-compose/cli-config-support-default-system-config.patch:
```diff
@@ -1,4 +1,4 @@
-From 6385df5e8c5c4f3eeff0962ce7a182e9aac68f2c Mon Sep 17 00:00:00 2001
```
Same.

quaresmajose replied:
have that on https://github.com/foundriesio/meta-lmp/pull/1444. fixed now

ricardosalveti commented on meta-lmp-base/recipes-containers/docker/docker-ce/0001-dynbinary-use-go-cross-compiler.patch:
```diff
@@ -1,114 +0,0 @@
-From 6c1d9dea0ef2000cdeec38b46d60b2db66160ed3 Mon Sep 17 00:00:00 2001
```
Same.

quaresmajose replied:
have that on https://github.com/foundriesio/meta-lmp/pull/1444. fixed now

ricardosalveti commented on meta-lmp-base/conf/distro/include/lmp.inc:
```diff
@@ -37,7 +37,6 @@ PREFERRED_PROVIDER_nativesdk-zlib ?= "nativesdk-zlib"
 PREFERRED_PROVIDER_qemu-native ?= "qemu-native"
 PREFERRED_PROVIDER_qemu-helper-native ?= "qemu-helper-native"
 PREFERRED_PROVIDER_nativesdk-qemu ?= "nativesdk-qemu"
-PREFERRED_PROVIDER_virtual/docker = "docker-ce"
```
Please explain why in your commit message.

quaresmajose replied:
have that on https://github.com/foundriesio/meta-lmp/pull/1444. fixed now

ricardosalveti commented on meta-lmp-base/dynamic-layers/tpm-layer/recipes-tpm2/tpm2-tools/tpm2-tools_5.5.bb:
```diff
@@ -1,18 +0,0 @@
-SUMMARY = "Tools for TPM2."
```
Why the revert? Please explain as well.

quaresmajose replied:
fixed

ricardosalveti commented on meta-lmp-base/dynamic-layers/tpm-layer/recipes-tpm2/tpm2-abrmd/files/tpm2-abrmd-init.sh:
```diff
@@ -1,65 +0,0 @@
-#!/bin/sh
-
```
Same.

ricardosalveti commented on meta-lmp-base/dynamic-layers/tpm-layer/recipes-tpm2/tpm2-tss/tpm2-tss/fixup_hosttools.patch:
```diff
@@ -1,29 +0,0 @@
-revert configure: add checks for all tools used by make install
-
```
Same.

quaresmajose replied:
fixed

ricardosalveti commented on meta-lmp-base/dynamic-layers/tpm-layer/recipes-tpm2/tpm2-pkcs11/tpm2-pkcs11_1.9.0.bb:
```diff
@@ -1,47 +0,0 @@
-SUMMARY = "A PKCS#11 interface for TPM2 hardware"
-DESCRIPTION = "PKCS #11 is a Public-Key Cryptography Standard that defines a standard method to access cryptographic services from tokens/ devices such as hardware security modules (HSM), smart cards, etc. In this project we intend to use a TPM2 device as the cryptographic token."
```
Same.

ricardosalveti commented on meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:
```diff
@@ -151,8 +151,8 @@ OSTREE_KERNEL_ARGS:mx6ull-generic-bsp ?= "console=tty1 console=ttymxc0,115200 ${
 OSTREE_KERNEL_ARGS:mx7d-generic-bsp ?= "console=tty1 console=ttymxc0,115200 ${OSTREE_KERNEL_ARGS_COMMON}"
 OSTREE_KERNEL_ARGS:mx7ulp-generic-bsp ?= "console=tty1 console=ttyLP0,115200 ${OSTREE_KERNEL_ARGS_COMMON}"
 
-PREFERRED_VERSION_meson ?= "0.61.3"
```
Please explain the reason for the revert in your commit message.

quaresmajose replied:
fixed

ricardosalveti commented on meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:
```diff
@@ -151,9 +151,6 @@ OSTREE_KERNEL_ARGS:mx6ull-generic-bsp ?= "console=tty1 console=ttymxc0,115200 ${
 OSTREE_KERNEL_ARGS:mx7d-generic-bsp ?= "console=tty1 console=ttymxc0,115200 ${OSTREE_KERNEL_ARGS_COMMON}"
 OSTREE_KERNEL_ARGS:mx7ulp-generic-bsp ?= "console=tty1 console=ttyLP0,115200 ${OSTREE_KERNEL_ARGS_COMMON}"
 
-PREFERRED_VERSION_meson:imx8-nxp-bsp ?= "0.63.3.imx"
```
Same.

quaresmajose replied:
fixed

ricardosalveti commented:
Mostly fine, but please improve the commit message saying why things were removed, renamed or reverted.

quaresmajose commented on meta-lmp-base/recipes-security/optee/files/0001-FIO-fromtree-core-mm-Fix-conflicting-types-due-to-en.patch:
```diff
@@ -1,33 +0,0 @@
-From c95bf46abd1ee18d555e10f557c9406d99fcad43 Mon Sep 17 00:00:00 2001
```
both are integrated into the FIO main tree. fixed now

quaresmajose commented on meta-lmp-base/recipes-containers/runc/files/0001-Makefile-respect-GOBUILDFLAGS-for-runc-and-remove-re.patch:
```diff
@@ -1,28 +0,0 @@
-From 0fe50d2ca4517f5e3070585040f35ace413acd44 Mon Sep 17 00:00:00 2001
```
both are integrated into the FIO main tree. fixed now

quaresmajose commented:
> Mostly fine, but please improve the commit message saying why things were removed, renamed or reverted.

Suggestions added and fixed.
Thanks for the review.

angolini commented on meta-lmp-base/conf/distro/lmp-base.conf:
I understand we can now enable the minidebuginfo, something that has a complex configuration setting defined in meta, but we only use it. In this case, the commit message is misleading and makes me think I need to configure gdb for example.

On the other hand, I understand where it came from. I like you add the link to the YP commit

ricardosalveti replied:
Yeah, the commit message is a bit confusing, would be better to just point to the upstream commit saying we're doing the same for xyz reasons.

quaresmajose replied:
changed to explain what we add and for what propose

angolini commented on meta-lmp-base/recipes-extended/ostree/ostree_%.bbappend:
I understand we are reverting this commit as a consequence of  `PACKAGECONFIG:remove = "static"` in the previous commit

ricardosalveti replied:
I still have the impression that static will be required, but would need to validate at runtime.

quaresmajose replied:
> I understand we are reverting this commit as a consequence of PACKAGECONFIG:remove = "static" in the previous commit

Yes, the reason of the revert is the previous commit but I will mention that on message.

> I still have the impression that static will be required, but would need to validate at runtime.

The main reason of this was to support the composefs.

angolini commented on meta-lmp-base/recipes-sota/aktualizr/aktualizr-fioefi-env-rollback.bb:
```diff
@@ -0,0 +1,23 @@
+
+SRC_URI = "file://sota-fioefi-env.toml"
+
+PV = "1.0"
```
Do we really need to set PV explicitly inside the recipe? I would prefer to use it on filename

ricardosalveti replied:
Yeah, better to name it with _1.0 instead.

angolini commented on meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:
```diff
@@ -766,6 +766,7 @@ IMAGE_TEGRAFLASH_KERNEL:tegra = "${DEPLOY_DIR_IMAGE}/${KERNEL_IMAGETYPE}"
 WKS_FILE:tegra:sota ?= "efidisk-sota.wks.in"
 IMAGE_ROOTFS_SIZE:tegra = "65536"
 PREFERRED_PROVIDER_virtual/bootloader:tegra ?= "edk2-firmware-tegra"
+TEGRA_UEFI_CAPSULE_INSTALL_DIR:tegra = ""
```
Maybe it's a little too much to have a TEGRA variable name and a tegra override

angolini commented:
For me, it looks like the addition of `fioefi` as  `MACHINE_FEATURES` would not be need, as we could use `efi` instead. However, I understand we decided to do so to standardize with `fiovb`

ricardosalveti commented on meta-lmp-base/recipes-containers/runc/runc-opencontainers_git.bb:
```diff
@@ -1,10 +0,0 @@
-include recipes-containers/runc/runc.inc
-
-SRCREV = "51d5e94601ceffbbd85688df1c928ecccbfa4685"
```
Fine to drop runc-opencontainers and containerd-opencontainers here, but my goal when dealing with these recipes in lmp was to avoid using rcs and beta releases from upstream, so just to keep in mind as we do the remaining updates. These components are critical to LmP, so better to use the released / stable versions instead (and we also need to validate if docker itself was tested and released with the same base versions here).

quaresmajose replied:
I believe that in the LTS scarthgap branch of meta-virt we will only have stable versions.
However, for some time to come we will probably need to update these same stable. Maybe when we go back to following what's in the master again, there may be rcs and beta releases.
but agree that it is an important point to keep in mind.

ricardosalveti commented on meta-lmp-base/recipes-sota/fioefi/fioefi_0.1.bb:
```diff
@@ -3,6 +3,7 @@ LICENSE = "MIT"
 LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
 
 SRC_URI = "file://fioefi.sh.in"
+RDEPENDS:${PN}:append = " lmp-boot-firmware"
```
Here I generally prefer += instead of using append, which is harder to override if needed.

ricardosalveti commented:
> For me, it looks like the addition of `fioefi` as `MACHINE_FEATURES` would not be need, as we could use `efi` instead. However, I understand we decided to do so to standardize with `fiovb`

Yeah, I agree that in general 'efi' would be enough to imply fioefi here. (or at least automatically add fioefi when efi is available).

ricardosalveti commented:
General comment about fioefi, probably better to separate our own development done in scarthgap as part of another PR, as they are not strictly related to our scarthgap updating work (they are extras).

quaresmajose commented on meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:
fixed

quaresmajose commented:
> General comment about fioefi, probably better to separate our own development done in scarthgap as part of another PR, as they are not strictly related to our scarthgap updating work (they are extras).

I will move the fioefi patchset again to the scarthgap-next branch that contains what is still pending

quaresmajose commented:
Thanks for the review

quaresmajose commented:
> > > General comment about fioefi, probably better to separate our own development done in scarthgap as part of another PR, as they are not strictly related to our scarthgap updating work (they are extras).
> > 
> > 
> > I will move the fioefi patchset again to the scarthgap-next branch that contains what is still pending
> 
> When you create it, please add me to the reviewers. I'm afraid github inbox don't notify me, and it's still pending in my mind what level of documentation the change requires. Thanks

I have already move it to https://github.com/foundriesio/meta-lmp/commits/scarthgap-next
IMO I will be better if @igoropaniuk can submit this again when we have the main branch in a better state

ricardosalveti reviewed (approved):
LGTM
